src/main/java/com/karolina/tuesdaystasks/peselnumber/PeselSexExtractor.java

```diff
@@ -0,0 +1,25 @@
+package com.karolina.tuesdaystasks.peselnumber;
+
+import static com.karolina.tuesdaystasks.peselnumber.util.StringUtils.getDigitAtIndexFrom;
+
+public class PeselSexExtractor implements DataExtractor {
+
+    @Override
+    public String extract(String encodedData) {
+        int controlDigit = getDigitAtIndexFrom(9, encodedData);
+        if (controlDigit % 2 == 0) {
+            return Gender.WOMAN.toString();
+        }
+        return Gender.MAN.toString();

```

Comment:
jakby DataExtractor był generyczny, to mogłabyś tutaj zwracać `Gender` normalnie, a teraz nadal jest nieznanego formatu String